src/features/textEditor/components/tipTapNodes/notionBlock/NotionBlockNode.ts

```diff
@@ -0,0 +1,327 @@
+    placeholder.innerHTML = 'Empty toggle. Click or drop blocks inside.';
+    placeholder.style.color = 'hsl(var(--color-muted))';
+    placeholder.classList.add('empty-collapsible-placeholder');
+
+    return placeholder;
+  });
+}
+
+function getCollapsiblePlaceholderDecorations(state: EditorState): Decoration[] {
+  const decorations: Decoration[] = [];
+  state.doc.descendants((node, pos) => {
+    if (node.type.name !== NotionBlockNode.name) {
+      return false;
+    }
+    if (node.attrs.type !== 'collapsible') {
+      return;
+    }
+
+    const isEmpty = node.childCount === 1;
+    const isCollapsed = !!node.attrs.collapsed;
+    if (isEmpty && !isCollapsed) {
+      // Add the decoration after the paragraph (...|<notionBlock><p>...</p>... --> ...<notionBlock><p>...</p>|...)
+      decorations.push(createEmptyCollapsiblePlaceholderWidget(pos + node.child(0).nodeSize + 1));
+    }
+  });
+  return decorations;
+}
+
+export const collapsiblePluginKey = new PluginKey<DecorationSet>('collapsibleNodes');
+const collapsiblePlugin = new Plugin({
+  key: collapsiblePluginKey,
+  state: {
+    init: (_config, state) => DecorationSet.create(state.doc, getCollapsibleArrowDecorations(state)),
+    apply: (tr, value, _oldState, newState) => {
+      if (!tr.docChanged || tr.getMeta(collapsiblePluginKey)) {
+        return value;
+      }
+      return DecorationSet.create(tr.doc, getCollapsibleArrowDecorations(newState));
+    },
+  },
+  props: {
+    decorations(state) {
+      return this.getState(state);
+    },
+    handleClick: (view, pos, event: MouseEvent) => {
+      const target = event.target as HTMLElement;
+      const toggleNodeCollapsed = () =>
+        toggleCollapsed({ pos: pos - 2, view, tr: view.state.tr, dispatch: (tr: Transaction) => view.dispatch(tr) });
+      // We need this because user can click on the svg polygon element, that is not an instance of HTMLElement
+      if (target.parentElement?.parentElement?.className.includes('collapsible-arrow')) {
+        toggleNodeCollapsed();
+      } else if (target.parentElement?.className.includes('collapsible-arrow')) {
+        toggleNodeCollapsed();
+      } else if (target.className.includes('collapsible-arrow')) {
+        toggleNodeCollapsed();
+      } else if (target.className.includes('empty-collapsible-placeholder')) {
+        const { tr } = view.state;
+        const schemaNodes = view.state.schema.nodes;
+        view.dispatch(tr.insert(pos, schemaNodes[NotionBlockNode.name].create(null, schemaNodes.paragraph.create())));
+      }
+    },
+  },
+});
+export const collapsiblePlaceholderPluginKey = new PluginKey<DecorationSet>('collapsibleNodes');
+const collapsiblePlaceholderPlugin = new Plugin({
+  key: collapsiblePlaceholderPluginKey,
+  state: {
+    init: (_config, state) => DecorationSet.create(state.doc, getCollapsiblePlaceholderDecorations(state)),
+    apply: (tr, value, _oldState, newState) => {
+      if (!tr.docChanged) {
```

Comment:
Why does this wind up looking so different than `placeholderPlugin`? They seem to be doing similar things (adding decorations for empty blocks) but this one has some checks (`tr.docChanged`) that `placeholderPlugin` lacks and the `getDecorations` code looks much more complicated.

Reply:
Both plugins are actually quite different: `placeholderPlugin` adds a placeholder when the cursor is in an empty block, whereas `collapsiblePlaceholderPluginKey` adds a placeholder to all empty collapsible blocks.
So the former just needs to check that selection is empty and in an empty block and then add the decoration at the cursor position. The latter needs to iterate through the document to find empty collapsible blocks (this is why `getDecorations` is more complicated) ; and it checks `tr.docChanged` to prevent recomputing decorations when the doc has not changed.